lib/pp_loc.ml

```diff
@@ -366,6 +402,23 @@ let pp
     (ppf: Format.formatter)
     (locs: loc list)
   =
+
+  (* convert locations so that the positions are full lexing positions *)
+  let locs = match locs with
+    | [] -> []
+    | _ ->
+      let input_raw, close_input = Input.open_raw input in
+      let locs =
+        List.map
+          (fun (start_pos,end_pos) ->
+             Position.to_full input_raw start_pos,
+             Position.to_full input_raw end_pos)
+          locs
+      in
+      close_input ();
+      locs
+  in
```

Comment:
It is a bit subtle why this pattern works (opening, closing, opening again afterwards).
If `input` has been created from an `in_channel`, then `Input.open_raw` and `close_input` are no-ops, in which case converting the locations will move the cursor to the end of the file. This is fine because `lines_around` afterwards takes care of `seek`ing before reading anything, but a bit counter-intuitive (it reads a bit like open/close would reset the cursor, but they don't).

I think it would be more natural to only do only one `Input.open_raw` and `close_input` for the whole `pp` invocation.

Reply:
ok, I think I changed it in a reasonable way, but it'd be good to have a few tests I think

Reply:
done